From 7c37099d783f8fb83969c26e9fcb5e891f736a90 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gunnar@beutner.name>
Date: Wed, 12 Oct 2011 12:49:18 +0200
Subject: [PATCH] vn_rdwr() didn't properly advance the file position

This would cause problems when using 'zfs send' with a file as the
target (rather than a pipe or a socket as is usually the case) as
for each write the destination offset in the file would be 0.
---
 module/spl/spl-vnode.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/module/spl/spl-vnode.c b/module/spl/spl-vnode.c
index 1b5cc5a..3d0bb5a 100644
--- a/module/spl/spl-vnode.c
+++ b/module/spl/spl-vnode.c
@@ -248,6 +248,8 @@ vn_rdwr(uio_rw_t uio, vnode_t *vp, void *addr, ssize_t len, offset_t off,
 
 	set_fs(saved_fs);
 
+	fp->f_pos = offset;
+
 	if (rc < 0)
 		SRETURN(-rc);
 
-- 
1.7.5.4

